File: src/main/java-graphql/io/github/graphql/java/model/RepositoryInfoTO.java
package io.github.graphql.java.model;


@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-07T19:46:14+0800"
)
public interface RepositoryInfoTO {

    @javax.validation.constraints.NotNull
    String getCreatedAt();

    String getDescription();

    @javax.validation.constraints.NotNull
    String getDescriptionHTML();

    int getForkCount();

    boolean getHasIssuesEnabled();

    boolean getHasWikiEnabled();

    String getHomepageUrl();

    boolean getIsArchived();

    boolean getIsFork();

    boolean getIsLocked();

    boolean getIsMirror();

    boolean getIsPrivate();

    boolean getIsTemplate();

    LicenseTO getLicenseInfo();

    RepositoryLockReasonTO getLockReason();

    String getMirrorUrl();

    @javax.validation.constraints.NotNull
    String getName();

    @javax.validation.constraints.NotNull
    String getNameWithOwner();

    @javax.validation.constraints.NotNull
    String getOpenGraphImageUrl();

    @javax.validation.constraints.NotNull
    RepositoryOwnerTO getOwner();

    String getPushedAt();

    @javax.validation.constraints.NotNull
    String getResourcePath();

    @javax.validation.constraints.NotNull
    String getUpdatedAt();

    @javax.validation.constraints.NotNull
    String getUrl();

    boolean getUsesCustomOpenGraphImage();

}

File: src/main/scala-graphql/io/github/graphql/model/RepositoryInfoTO.scala
package io.github.graphql.model

import RepositoryLockReasonTO._

@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-06T19:48:00+0800"
)
trait RepositoryInfoTO {

    @javax.validation.constraints.NotNull
    val createdAt: String

    val description: String

    @javax.validation.constraints.NotNull
    val descriptionHTML: String

    val forkCount: Int

    val hasIssuesEnabled: Boolean

    val hasWikiEnabled: Boolean

    val homepageUrl: String

    val isArchived: Boolean

    val isFork: Boolean

    val isLocked: Boolean

    val isMirror: Boolean

    val isPrivate: Boolean

    val isTemplate: Boolean

    val licenseInfo: LicenseTO

    val lockReason: RepositoryLockReasonTO

    val mirrorUrl: String

    @javax.validation.constraints.NotNull
    val name: String

    @javax.validation.constraints.NotNull
    val nameWithOwner: String

    @javax.validation.constraints.NotNull
    val openGraphImageUrl: String

    @javax.validation.constraints.NotNull
    val owner: RepositoryOwnerTO

    val pushedAt: String

    @javax.validation.constraints.NotNull
    val resourcePath: String

    @javax.validation.constraints.NotNull
    val updatedAt: String

    @javax.validation.constraints.NotNull
    val url: String

    val usesCustomOpenGraphImage: Boolean

}
